src/consensus/consensus.rs

```diff
@@ -314,56 +321,61 @@ pub enum BlockProposerError {
 
     #[error(transparent)]
     RpcResponseError(#[from] tonic::Status),
+
+    #[error(transparent)]
+    BlockStorageError(#[from] BlockStorageError),
 }
-#[derive(Default)]
+#[derive(Default, Clone)]
```

Comment:
Perhaps in a follow on PR we could move this to a separate file (and maybe some other big structs in here?)